Watching for file changes with StatReloader
Not Found: /
"GET / HTTP/1.1" 404 2171
"GET /firstapp/ HTTP/1.1" 200 746
"GET /firstapp/about/ HTTP/1.1" 200 1314
Not Found: /firstapp/page
"GET /firstapp/page HTTP/1.1" 404 2465
Watching for file changes with StatReloader
Not Found: /
"GET / HTTP/1.1" 404 2171
"GET /firstapp/ HTTP/1.1" 200 746
"GET /firstapp/about/ HTTP/1.1" 200 1314
"GET /firstapp/about/ HTTP/1.1" 200 1314
Not Found: /firstapp/page
"GET /firstapp/page HTTP/1.1" 404 2465
Watching for file changes with StatReloader
Not Found: /
"GET / HTTP/1.1" 404 2171
"GET /firstapp/about/ HTTP/1.1" 200 1314
"GET /firstapp/ HTTP/1.1" 200 746
"GET /firstapp/about/ HTTP/1.1" 200 1314
Not Found: /firstapp/page
"GET /firstapp/page HTTP/1.1" 404 2465
Watching for file changes with StatReloader
Not Found: /
"GET / HTTP/1.1" 404 2171
"GET /firstapp/about/ HTTP/1.1" 200 1314
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Not Found: /
"GET / HTTP/1.1" 404 2390
"GET /thirdapp HTTP/1.1" 301 0
"GET /thirdapp/ HTTP/1.1" 200 9
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 17, in client_orders
    context = {'client': client.name, 'order_base': orders_base}
AttributeError: 'NoneType' object has no attribute 'name'
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 67771
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 19, in client_orders
    return render(request, 'firstapp3/index.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 860, in do_for
    sequence = parser.compile_filter(bits[in_index + 1])
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 597, in compile_filter
    return FilterExpression(token, self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 698, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: '()' from 'orders_base.items()'
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 149286
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 19, in client_orders
    return render(request, 'firstapp3/index.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 515, in parse
    self.unclosed_block_tag(parse_until)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 578, in unclosed_block_tag
    raise self.error(token, msg)
django.template.exceptions.TemplateSyntaxError: Unclosed tag on line 9: 'for'. Looking for one of: empty, endfor.
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 147840
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Not Found: /
"GET / HTTP/1.1" 404 2390
"GET /thirdapp/ HTTP/1.1" 200 9
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 20, in client_orders
    return render(request, 'firstapp3/index.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 515, in parse
    self.unclosed_block_tag(parse_until)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 578, in unclosed_block_tag
    raise self.error(token, msg)
django.template.exceptions.TemplateSyntaxError: Unclosed tag on line 9: 'for'. Looking for one of: empty, endfor.
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 147685
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 21, in client_orders
    return render(request, 'firstapp3/index.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 515, in parse
    self.unclosed_block_tag(parse_until)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 578, in unclosed_block_tag
    raise self.error(token, msg)
django.template.exceptions.TemplateSyntaxError: Unclosed tag on line 9: 'for'. Looking for one of: empty, endfor.
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 147822
Watching for file changes with StatReloader
Not Found: /
"GET / HTTP/1.1" 404 2390
"GET /thirdapp/ HTTP/1.1" 200 9
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 21, in client_orders
    return render(request, 'firstapp3/index.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 515, in parse
    self.unclosed_block_tag(parse_until)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 578, in unclosed_block_tag
    raise self.error(token, msg)
django.template.exceptions.TemplateSyntaxError: Unclosed tag on line 9: 'for'. Looking for one of: empty, endfor.
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 151141
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 17, in client_orders
    orders_base.setdefault(str(order), list(products))
TypeError: 'ManyRelatedManager' object is not iterable
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 68856
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 17, in client_orders
    orders_base.setdefault(str(order), list(*products))
TypeError: list() argument after * must be an iterable, not ManyRelatedManager
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 68914
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 17, in client_orders
    orders_base.setdefault(order.pk, *products)
TypeError: Value after * must be an iterable, not ManyRelatedManager
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 68858
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 22, in client_orders
    return render(request, 'firstapp3/index.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 515, in parse
    self.unclosed_block_tag(parse_until)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 578, in unclosed_block_tag
    raise self.error(token, msg)
django.template.exceptions.TemplateSyntaxError: Unclosed tag on line 9: 'for'. Looking for one of: empty, endfor.
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 150566
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 22, in client_orders
    return render(request, 'firstapp3/index.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 515, in parse
    self.unclosed_block_tag(parse_until)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 578, in unclosed_block_tag
    raise self.error(token, msg)
django.template.exceptions.TemplateSyntaxError: Unclosed tag on line 9: 'for'. Looking for one of: empty, endfor.
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 150567
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 22, in client_orders
    return render(request, 'firstapp3/index.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 515, in parse
    self.unclosed_block_tag(parse_until)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 578, in unclosed_block_tag
    raise self.error(token, msg)
django.template.exceptions.TemplateSyntaxError: Unclosed tag on line 9: 'for'. Looking for one of: empty, endfor.
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 149315
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 22, in client_orders
    return render(request, 'firstapp3/index.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 515, in parse
    self.unclosed_block_tag(parse_until)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 578, in unclosed_block_tag
    raise self.error(token, msg)
django.template.exceptions.TemplateSyntaxError: Unclosed tag on line 9: 'for'. Looking for one of: empty, endfor.
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 149100
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\models.py changed, reloading.
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 161
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 452
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 22, in client_orders
    return render(request, 'firstapp3/index.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 847, in do_for
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: 'for' statements should use the format 'for x in y': for product in {{ products }}
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 156862
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 597
Internal Server Error: /thirdapp/clientorders/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 22, in client_orders
    return render(request, 'firstapp3/index.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 481, in parse
    raise self.error(token, e)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 479, in parse
    filter_expression = self.compile_filter(token.contents)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 597, in compile_filter
    return FilterExpression(token, self)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 698, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: '*products' from '*products'
"GET /thirdapp/clientorders/1/ HTTP/1.1" 500 161664
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 474
"GET /thirdapp/clientorders/2/ HTTP/1.1" 200 273
"GET /thirdapp/clientorders/3/ HTTP/1.1" 200 376
Watching for file changes with StatReloader
"GET /thirdapp/clientorders/3/ HTTP/1.1" 200 376
"GET /thirdapp/clientorders/1/ HTTP/1.1" 200 474
Watching for file changes with StatReloader
Not Found: /
"GET / HTTP/1.1" 404 2390
"GET /thirdapp/ HTTP/1.1" 200 9
Not Found: /thirdapp/periodorders/2/7/
"GET /thirdapp/periodorders/2/7/ HTTP/1.1" 404 3003
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\urls.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/periodorders/2/7/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 30, in client_orders_period
    orders = Order.objects.filter(customer=client).filter(creation_date__gt=(date.today() - timespan))
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\db\models\query.py", line 1476, in filter
    return self._filter_or_exclude(False, args, kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\db\models\query.py", line 1494, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\db\models\query.py", line 1501, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\db\models\sql\query.py", line 1613, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\db\models\sql\query.py", line 1645, in _add_q
    child_clause, needed_inner = self.build_filter(
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\db\models\sql\query.py", line 1495, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\db\models\sql\query.py", line 1307, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\db\models\sql\query.py", line 1772, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'creation_date' into field. Choices are: customer, customer_id, date_order, id, products, total_price
"GET /thirdapp/periodorders/2/7/ HTTP/1.1" 500 109528
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /thirdapp/periodorders/2/7/ HTTP/1.1" 200 273
"GET /thirdapp/periodorders/2/1/ HTTP/1.1" 200 273
Watching for file changes with StatReloader
"GET /thirdapp/periodorders/2/1/ HTTP/1.1" 200 229
"GET /thirdapp/periodorders/2/1/ HTTP/1.1" 200 229
Watching for file changes with StatReloader
"GET /thirdapp/periodorders/2/1/ HTTP/1.1" 200 389
Watching for file changes with StatReloader
"GET /thirdapp/periodorders/2/1/ HTTP/1.1" 200 417
Watching for file changes with StatReloader
"GET /thirdapp/periodorders/2/1/ HTTP/1.1" 200 545
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /thirdapp/periodorders/2/1/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp3\views.py", line 40, in client_orders_period
    return render(request, 'firstapp3/orders.html', context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 326, in render
    return nodelist.render(context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\defaulttags.py", line 197, in render
    values = list(values)
TypeError: 'ManyRelatedManager' object is not iterable
"GET /thirdapp/periodorders/2/1/ HTTP/1.1" 500 150236
Watching for file changes with StatReloader
"GET /thirdapp/periodorders/2/1/ HTTP/1.1" 200 698
Watching for file changes with StatReloader
"GET /thirdapp/periodorders/2/1/ HTTP/1.1" 200 742
"GET /thirdapp/periodorders/2/7/ HTTP/1.1" 200 742
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 746
"GET /app2 HTTP/1.1" 301 0
"GET /app2/ HTTP/1.1" 200 14
"GET /app3/ HTTP/1.1" 200 14
Internal Server Error: /app4/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp4\views.py", line 5, in index
    return render(request, 'firstapp4/index.html')
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: firstapp4/index.html
"GET /app4/ HTTP/1.1" 500 81094
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\hwproject\settings.py changed, reloading.
Watching for file changes with StatReloader
"GET /app4/ HTTP/1.1" 200 150
"GET /app5/ HTTP/1.1" 200 150
"GET /app6/ HTTP/1.1" 200 150
"GET /app3/ HTTP/1.1" 200 14
Not Found: /app2/sdf
"GET /app2/sdf HTTP/1.1" 404 3018
Not Found: /app2/1
"GET /app2/1 HTTP/1.1" 404 3012
"GET /app2/ HTTP/1.1" 200 14
Not Found: /app/about
"GET /app/about HTTP/1.1" 404 2960
Not Found: /app/about/
"GET /app/about/ HTTP/1.1" 404 2963
"GET /about/ HTTP/1.1" 200 1314
"GET / HTTP/1.1" 200 746
"GET /about/ HTTP/1.1" 200 1314
"GET /app2/ HTTP/1.1" 200 14
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp2\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /app2/ HTTP/1.1" 200 119
"GET /app3/ HTTP/1.1" 200 14
Not Found: /app3/client
"GET /app3/client HTTP/1.1" 404 3476
"GET /app3/clientorders/1 HTTP/1.1" 301 0
"GET /app3/clientorders/1/ HTTP/1.1" 200 474
Not Found: /app3/cliento
"GET /app3/cliento HTTP/1.1" 404 3479
"GET /app3/periodorders/1/7 HTTP/1.1" 301 0
"GET /app3/periodorders/1/7/ HTTP/1.1" 200 747
"GET /app3/periodorders/2/7/ HTTP/1.1" 200 742
"GET /app4 HTTP/1.1" 301 0
"GET /app4/ HTTP/1.1" 200 150
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 746
"GET /app4/ HTTP/1.1" 200 352
"GET /app4/img_up HTTP/1.1" 301 0
"GET /app4/img_up/ HTTP/1.1" 200 352
"GET /app4/img_up/ HTTP/1.1" 200 352
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp4\urls.py changed, reloading.
Watching for file changes with StatReloader
"GET /app4/img_up/ HTTP/1.1" 200 536
Internal Server Error: /app4/img_up/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp4\views.py", line 16, in image_upload
    if form.is_valid():
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 197, in is_valid
    return self.is_bound and not self.errors
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 192, in errors
    self.full_clean()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 327, in full_clean
    self._clean_fields()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 337, in _clean_fields
    value = field.clean(value, bf.initial)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\fields.py", line 689, in clean
    return super().clean(data)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\fields.py", line 204, in clean
    value = self.to_python(value)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\fields.py", line 716, in to_python
    from PIL import Image
ModuleNotFoundError: No module named 'PIL'
"POST /app4/img_up/ HTTP/1.1" 500 98066
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\hwproject\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\hwproject\settings.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /app4/img_up/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp4\views.py", line 16, in image_upload
    if form.is_valid():
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 197, in is_valid
    return self.is_bound and not self.errors
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 192, in errors
    self.full_clean()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 327, in full_clean
    self._clean_fields()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 337, in _clean_fields
    value = field.clean(value, bf.initial)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\fields.py", line 689, in clean
    return super().clean(data)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\fields.py", line 204, in clean
    value = self.to_python(value)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\fields.py", line 716, in to_python
    from PIL import Image
ModuleNotFoundError: No module named 'PIL'
"POST /app4/img_up/ HTTP/1.1" 500 98206
"GET /app4/img_up HTTP/1.1" 301 0
"GET /app4/img_up/ HTTP/1.1" 200 536
Internal Server Error: /app4/img_up/
Traceback (most recent call last):
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\hwproject\firstapp4\views.py", line 16, in image_upload
    if form.is_valid():
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 197, in is_valid
    return self.is_bound and not self.errors
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 192, in errors
    self.full_clean()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 327, in full_clean
    self._clean_fields()
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\forms.py", line 337, in _clean_fields
    value = field.clean(value, bf.initial)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\fields.py", line 689, in clean
    return super().clean(data)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\fields.py", line 204, in clean
    value = self.to_python(value)
  File "C:\Users\Lush\Documents\Learning\Цифровые профессии\Фреймворк Django\hw_project\venv\lib\site-packages\django\forms\fields.py", line 716, in to_python
    from PIL import Image
ModuleNotFoundError: No module named 'PIL'
"POST /app4/img_up/ HTTP/1.1" 500 98206
"GET /app4/img_up/ HTTP/1.1" 200 536
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 746
"GET /app4/img_up/ HTTP/1.1" 200 536
"POST /app4/img_up/ HTTP/1.1" 200 536
Watching for file changes with StatReloader
"GET /app4/prod_up/ HTTP/1.1" 200 1090
